src/EventStore.Core/Services/PersistentSubscription/PersistentSubscriptionService.cs

```diff
@@ -719,29 +791,6 @@ string user
 			);
 		}
 
-		private void RemoveSubscriptionConfig(string username, string eventSource, string groupName) {
```

Comment:
suggestion:
to avoid code repetition in the update & remove methods, could we keep these two methods and add a parameter named: `replaceBy`  ?

```
RemoveSubscriptionConfig(string username, string eventSource, string groupName, PersistentSubscriptionEntry replaceBy);
RemoveSubscription(string eventSource, string groupName, PersistentSubscription replaceBy)
```

when `replaceBy` is null, we call RemoveAt(index) and when it's not null, we can replace the value at the index with that subscription or config.

Reply:
the replacement null check would also need to be done for `_subscriptionsById.Remove(key);`

Reply:
for more clarity, maybe we could then rename `RemoveSubscriptionConfig` to `UpdateSubscriptionConfig` and `RemoveSubscription` to `UpdateSubscription`

Reply:
Thanks a lot Shaan for the suggestions. Changes updated

Reply:
thanks very much Kunal. I've made a few formatting suggestions and some additional refactoring suggestions for the creation part.